sway-core/src/semantic_analysis/ast_node/mod.rs

```diff
@@ -130,6 +136,54 @@ impl ty::TyAstNode {
     }
 }
 
+fn handle_item_trait_imports(
+    ctx: &mut TypeCheckContext<'_>,
+    engines: &Engines,
+    handler: &Handler,
+) -> Result<(), ErrorEmitted> {
+    let mut impls_to_insert = TraitMap::default();
+
+    let root_mod = &ctx.namespace().root().module;
+    let dst_mod = ctx.namespace.module(engines);
+
+    for (_, (_, src, decl)) in dst_mod.current_items().use_item_synonyms.iter() {
+        let src_mod = root_mod.lookup_submodule(handler, engines, src)?;
+
+        //  if this is an enum or struct or function, import its implementations
+        if let Ok(type_id) = decl.return_type(&Handler::default(), engines) {
+            impls_to_insert.extend(
+                src_mod
+                    .current_items()
+                    .implemented_traits
+                    .filter_by_type_item_import(type_id, engines),
+                engines,
+            );
+        }
+        // if this is a trait, import its implementations
+        let decl_span = decl.span(engines);
+        if matches!(decl, TyDecl::TraitDecl(_)) {
+            // TODO: we only import local impls from the source namespace
```

Comment:
Should we open an issue for this TODO and link it here?

Reply:
It wasn't added by me. The commit you link to moved that part of the code from sway-core/src/semantic_analysis/namespace/module.rs as part of a refactoring.

The TODO is ancient AFAIK.